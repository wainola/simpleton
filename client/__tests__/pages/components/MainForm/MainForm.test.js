const React = require('react');
const { shallow, mount } = require('enzyme');
const { MainForm } = require('../../../../pages/components/MainForm');
const { Input } = require('../../../../pages/components/Input');

let formFields;
let evt;
let shallowComp;
let mountComp;

describe('<MainForm />', () => {
  beforeEach(() => {
    formFields = [
      { type: 'text', name: 'nombre', placeholder: 'nombre', title: 'Nombre' },
      { type: 'text', name: 'apellido', placeholder: 'apellido', title: 'Apellido' },
      { type: 'email', name: 'email', placeholder: 'email', title: 'Email' },
      { type: 'phone', name: 'telefono', placeholder: 'telefono', title: 'Telefono' },
      { type: 'address', name: 'direccion', placeholder: 'direccion', title: 'Direccion' },
      { type: 'textarea', name: 'razon', placeholder: 'razon', title: 'Razon' }
    ];

    evt = {
      preventDefault: function() {},
      target: {
        name: undefined,
        value: undefined
      }
    };

    shallowComp = shallow(<MainForm formFields={formFields} />);

    mountComp = mount(<MainForm formFields={formFields} />);
  });

  it('Should render without errors', () => {
    expect(shallowComp).toMatchSnapshot();
  });
});
