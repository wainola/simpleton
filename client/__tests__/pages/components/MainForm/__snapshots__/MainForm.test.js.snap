// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainForm /> Should render without errors 1`] = `
<Container
  as="div"
  className="form-container"
  fluid={false}
>
  <Bootstrap(Row)>
    <Card
      as="div"
      body={false}
    >
      <CardTitle
        className="form-title"
      >
        Contáctenos
      </CardTitle>
      <CardBody>
        <Form
          as="form"
          inline={false}
          onSubmit={[Function]}
        >
          <FormGroup
            as="div"
          >
            <Input
              name="nombre"
              onChange={[Function]}
              placeholder="ingrese su nombre"
              title="Nombre"
              type="text"
            />
            <Input
              name="apellido"
              onChange={[Function]}
              placeholder="ingrese su su apellido"
              title="Apellido"
              type="text"
            />
            <Input
              name="email"
              onChange={[Function]}
              placeholder="ingrese su correo electrónico"
              title="Correo electónico"
              type="email"
            />
            <Input
              name="telefono"
              onChange={[Function]}
              placeholder="ingrese su número de teléfono"
              title="Teléfono"
              type="phone"
            />
            <Input
              name="razon"
              onChange={[Function]}
              placeholder="ingrese el motivo de su consulta"
              title="Motivo de consulta"
              type="textarea"
            />
            <Input
              name="direccion"
              onChange={[Function]}
              placeholder="ingrese su dirección"
              title="Dirección"
              type="adddress"
            />
          </FormGroup>
          <FormGroup
            as="div"
          >
            <Button
              active={false}
              disabled={false}
              type="submit"
              variant="success"
            >
              Enviar!
            </Button>
          </FormGroup>
        </Form>
      </CardBody>
    </Card>
  </Bootstrap(Row)>
</Container>
`;
